### PR TITLE
Add "MH ET LIVE ESP32 Devkit and Minikit" board Support

### PR DESCRIPTION
**Pull request by MHEtLive**

### CODE REVIEW

**Comment by MHEtLive** [comment]
@me-no-dev  Want to add board support,please. is it correct?

File: boards.txt [hunk 1]
```diff
@@ -671,6 +671,60 @@ nodemcu-32s.menu.UploadSpeed.512000.upload.speed=512000
 
 ##############################################################
 
+MH-ET-LIVE-ESP32DevKIT.name=MH ET LIVE ESP32DevKIT
+
+esp32.upload.tool=esptool
```
Comment (me-no-dev) [inline boards.txt]: can not use ```esp32``` as board id ;) already defined!

File: boards.txt [hunk 2]
```diff
@@ -671,6 +671,60 @@ nodemcu-32s.menu.UploadSpeed.512000.upload.speed=512000
...
+esp32.upload.maximum_size=1044464
+esp32.upload.maximum_data_size=294912
+esp32.upload.wait_for_upload_port=true
+
+esp32.serial.disableDTR=true
+esp32.serial.disableRTS=true
+
+esp32.build.mcu=esp32
+esp32.build.core=esp32
+esp32.build.variant=esp32
```
Comment (me-no-dev) [inline boards.txt]: variant need to match the folder name of the variant with pins_arduino.h

File: variants/MH ET LIVE ESP32Devkit/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#ifndef Pins_Arduino_h
```
Comment (me-no-dev) [inline variants/MH ET LIVE ESP32Devkit/pins_arduino.h]: use lower case letters for variant (MH ET...) no spaces also!

Comment (MHEtLive) [inline variants/MH ET LIVE ESP32Devkit/pins_arduino.h]: Thanks for your reply! The linkhere:http://forum.mhetlive.com

**Comment by MHEtLive** [comment]
@me-no-dev  Thanks for your reply! The linkhere:http://forum.mhetlive.com

**Comment by MHEtLive** [comment]
@me-no-dev Hello! I have changed some incorrect points . Is it  all right to  be added? Thanks!

**Comment by me-no-dev** [comment]
have you tested this in Arduino IDE?

**Comment by capedra** [comment]
@me-no-dev 

Not working on my MH ET LIVE ESP32 Minikit. Here's the output of esptool:
```

C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/esptool.exe --chip esp32 --port COM5 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xe000 C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/partitions/boot_app0.bin 0x1000 C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/sdk/bin/{build.boot}.bin 0x10000 C:\Users\APJ\AppData\Local\Temp\arduino_build_77328/SD_ESP32_mount.ino.bin 0x8000 C:\Users\APJ\AppData\Local\Temp\arduino_build_77328/SD_ESP32_mount.ino.partitions.bin

usage: esptool write_flash [-h] [--flash_freq {keep,40m,26m,20m,80m}]
                           [--flash_mode {keep,qio,qout,dio,dout}]
                           [--flash_size FLASH_SIZE]
                           [--spi-connection SPI_CONNECTION] [--no-progress]
                           [--verify] [--compress | --no-compress]
                           <address> <filename> [<address> <filename> ...]

esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 'C:\\Users\\APJ\\Documents\\Arduino\\hardware\\espressif\\esp32/tools/sdk/bin/{build.boot}.bin'

esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 'C:\\Users\\APJ\\Documents\\Arduino\\hardware\\espressif\\esp32/tools/sdk/bin/{build.boot}.bin'

```

**Comment by capedra** [comment]
@me-no-dev 
Hello Hristo, nice to be here talking to you. Esptool is not working with these boards.

**Comment by MHEtLive** [comment]
@me-no-dev  Yes,i have tested in Arduino IDE? And make the guide in forum. Forum here:forum.mhetlive.com/category/2/devkit-minikit-testing

**Comment by capedra** [comment]
@me-no-dev I did that but the problem persists.

**Comment by me-no-dev** [comment]
@capedra see my comments above ;) one new line needs to be added to each board definition

**Comment by capedra** [comment]
@igrr 
@me-no-dev what comment? I made all these 3 changes on boards.txt file and also the variant files for both boards. Maybe the problem resides on the uppercase letters "MH-ET-LIVE-ESP32DevKIT" and "MH-ET-LIVE-ESP32MiniKit" located on boards.txt? I could test that. But I think that it's an issue on esptool.

**Comment by me-no-dev** [comment]
@capedra it is NOT an issue with esptool! it's an issue with the configuration!
add the following lines to the boards, reboot arduino IDE and try again!
```
MH-ET-LIVE-ESP32DevKIT.build.boot=bootloader
MH-ET-LIVE-ESP32MiniKIT.build.boot=bootloader
```

**Comment by capedra** [comment]
@me-no-dev 
thank you very much! I couldn't find that comment anywhere. I'm going to test it right now.
after that, I'm going back to my SD card issue on MH-ET Live Mini, which is not being mounted.

**Comment by capedra** [comment]
well, sometimes it mounts, sometimes it doesn't.

**Comment by capedra** [comment]
@igrr 
@me-no-dev same issue persists:
```
C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/esptool.exe --chip esp32 --port COM5 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xe000 C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/partitions/boot_app0.bin 0x1000 C:\Users\APJ\Documents\Arduino\hardware\espressif\esp32/tools/sdk/bin/{build.boot}.bin 0x10000 C:\Users\APJ\AppData\Local\Temp\arduino_build_811100/mySDLib_ESP32.ino.bin 0x8000 C:\Users\APJ\AppData\Local\Temp\arduino_build_811100/mySDLib_ESP32.ino.partitions.bin 


usage: esptool write_flash [-h] [--flash_freq {keep,40m,26m,20m,80m}]
                           [--flash_mode {keep,qio,qout,dio,dout}]
                           [--flash_size FLASH_SIZE]
                           [--spi-connection SPI_CONNECTION] [--no-progress]
                           [--verify] [--compress | --no-compress]
                           <address> <filename> [<address> <filename> ...]

esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 'C:\\Users\\APJ\\Documents\\Arduino\\hardware\\espressif\\esp32/tools/sdk/bin/{build.boot}.bin'

esptool write_flash: error: argument <address> <filename>: [Errno 2] No such file or directory: 'C:\\Users\\APJ\\Documents\\Arduino\\hardware\\espressif\\esp32/tools/sdk/bin/{build.boot}.bin'

```

**Comment by capedra** [comment]
Typo:
```
MH-ET-LIVE-ESP32MiniKit.build.boot=bootloader
```

**Comment by MHEtLive** [comment]
@capedra  I also got the issue you mentioned about the SDcard in my test just now. I will try to work out why.

**Comment by capedra** [comment]
@me-no-dev 
thank you. the configured board is working well now.
@MHEtLive thank you. I really need the SD card working but it doesn't mount anymore. Please help.
also, add the following two lines at the boards.txt file:
```
MH-ET-LIVE-ESP32DevKIT.build.boot=bootloader
MH-ET-LIVE-ESP32MiniKit.build.boot=bootloader
```

**Comment by capedra** [comment]
@MHEtLive Card Mount Failed

**Comment by MHEtLive** [comment]
I wonder if it is similar to this issue bellow. https://github.com/espressif/arduino-esp32/issues/272

**Comment by capedra** [comment]
@MHEtLive I already tested it with two different SD card modules but the problem persists.

**Comment by MHEtLive** [comment]
@capedra @me-no-dev 
I also tested it with different mainboards,32thing,doit esp32 and mhesp32devkits.However i found that  "Card Mount Failed" appeared nearly a third of resting times, with all these boards.

**Comment by capedra** [comment]
@MHEtLive @me-no-dev It's probably some hardware issue of ESP-32. Because it was working just before I pressed the reset button of the board.

**Comment by capedra** [comment]
@MHEtLive My MH-ET Live MiniKit just died. Could you please send me a new one? And I just figured it out something wrong on the official diagram. Thanks!

**Comment by MHEtLive** [comment]
@capedra Hello.What is wrong on the official diagram? And could you please tell me what`s  happened with your minikit and where did you buy it ? BTW,you can contact us on the mherlive forum.

**Comment by capedra** [comment]
@MHEtLive Hi. RXD pin is actually TXD pin and vice-versa. I tested it here. Actually, the ESP32 didn't die but the female micro-USB connector broke up very easy! Also, I can't flash it anymore because the connector is broken, and I can't initialize any SD cards on Arduino IDE or even MicroPython.

**Comment by MHEtLive** [comment]
@capedra Hello capedra.OK.The first issue may just a mistake that i haven`t point out is it OUTPUT/INPUT RXD/TXD on the board. The issue about the usb would be solved with a new version.Could you give me your email, In order to do more things for you?

**Comment by capedra** [comment]
@MHEtLive Sure. Well, I can't program the board using these pins. In other words, it's not letting me flash any new firmware using these pins (RXD/TXD). My e-mail is capedra@gmail.com. Thanks!

**Comment by MHEtLive** [comment]
@me-no-dev @reaper7 Thank you so much.

**Pull request closed without merging**